raiden_contracts/data/source/raiden/TokenNetwork.sol

```diff
@@ -1504,14 +1522,41 @@ contract TokenNetwork is Utils {
         view
         returns (bool)
     {
+        if (settle_input.claim.total_amount == 0) {
+            // Empty claims may be created at will without valid signature.
+            // They serve as a dummy value when no claim exist, since solidity
+            // does not allow us to pass a null value in that case.
+            return true;
+        }
         return (
             settle_input.claim.owner == settle_input.participant &&
-            settle_input.claim.partner == partner
-            // TODO: virtual channel fixme
-            // && claim_signer == recoverAddressFromClaim(settle_input.claim)
+            settle_input.claim.partner == partner &&
+            claim_signer == recoverAddressFromClaim(settle_input.claim)
         );
     }
 
+    function recoverAddressFromClaim(
+        Claim memory claim
+    )
+        internal view
+        returns (address signer_address)
+    {
+        // Length of the actual message: 20 + 32 + 20 + 20 + 32
```

Comment:
Is this correct? it looks like it is missing some fields

Reply:
The `signature_prefix` and  `message_length` are not fields that are counted towards the message length. If you don't count those, the number matches up.